src/components/forms/SoftRequiredErrors/utils.ts

```diff
@@ -0,0 +1,230 @@
+   *
+   * Each key, from the root to the component, is joined together in a period seperated
+   * string. For children of `editgrid` and `columns` components, the children index is
+   * included in the path.
+   *
+   * Example:
+   * Lets assume you have a columns component with the key "cars", which has a
+   * soft-required textfield in its first column with the key "brand".
+   *
+   * In this situation, the `pathToComponent` will be: "cars.0.brand".
+   */
+  pathToComponent: string;
+  /**
+   * The label to show for the "missing field" entry. This is typically the label of the
+   * empty soft-required component.
+   *
+   * When the missing field is located inside a `editgrid` or `fieldset` component, the
+   * labels of their parent components is also shown. For `editgrid` components, the
+   * groupLabel will also be included. Using the ">" character, we separate the labels to
+   * show the relationship in the form.
+   *
+   * Example:
+   * Lets assume you have a fieldset with the label "Car", which has a soft-required
+   * textfield with the label "Brand".
+   *
+   * In this situation, the `label` will be: "Car > Brand".
+   */
+  label: string;
+}
+
+const LABEL_SEPARATOR = ' > ';
+
+export const getMissingFields = (
+  SoftRequiredComponents: AnyComponentSchema[],
+  values: object,
+  getRegistryEntry: GetRegistryEntry,
+  keyPrefix: string[] = [],
+  labelPrefix: string[] = []
+): MissingFields[] => {
+  const missingFields: MissingFields[] = [];
+  for (const component of SoftRequiredComponents) {
+    const registry = getRegistryEntry(component);
+    const pathParts = [...keyPrefix, component.key];
+    const labelParts = [...labelPrefix, getIn(component, 'label', '')].filter(
+      label => label !== ''
+    );
+
+    const isEmpty = registry?.isEmpty;
+    const componentValue = getIn(values, pathParts.join('.'));
+
+    const isComponentEmpty = () => {
+      const isFalsy = componentValue == null || componentValue.length === 0;
+
+      return isEmpty !== undefined ? isEmpty(component, componentValue) : isFalsy;
+    };
+
+    switch (component.type) {
+      case 'fieldset':
+        missingFields.push(
+          ...getMissingFields(component.components, values, getRegistryEntry, pathParts, labelParts)
+        );
+        break;
+
+      case 'columns':
+        let columnIndex = 0;
+        for (const column of component.columns) {
+          if (column.components.length > 0) {
+            // Only check/add columns that actually have soft-required components to the
+            // missing list. If the soft-required fields aren't missing, nothing will be
+            // added.
+            missingFields.push(
+              ...getMissingFields(
+                column.components,
+                values,
+                getRegistryEntry,
+                [...pathParts, columnIndex.toString()],
+                [...labelParts]
+              )
+            );
+          }
+          columnIndex++;
+        }
```

Comment:
I'd reduce over the columns here - I think my earlier remark here about the column index being accessed in the values is wrong too - I suspect it breaks the `values` access in this form.